observer/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import "fmt"
+
+type Observable interface {
+	Add(observer Observer)
+	Remove(observer Observer)
+	Notify()
+
+	GetTemperature() float32
+	SetTemperature(temperature float32)
+}
+
+type WeatherStation struct {
+	temperature float32
+	observers   []Observer
+}
+
+func (ws *WeatherStation) Add(observer Observer) {
+	fmt.Println("Adding observer", observer.GetID())
+	ws.observers = append(ws.observers, observer)
+}
+
+func (ws *WeatherStation) Remove(observer Observer) {
+	fmt.Println("Removing", observer.GetID())
+	for index, o := range ws.observers {
+		if o.GetID() == observer.GetID() {
+			ws.observers = append(ws.observers[:index], ws.observers[index+1:]...)
+			break
+		}
+	}
+}
+
+func (ws *WeatherStation) Notify() {
+	for _, observer := range ws.observers {
+		observer.Update()
+	}
+}
+
+func (ws *WeatherStation) GetTemperature() float32 {
+	return ws.temperature
+}
+
+func (ws *WeatherStation) SetTemperature(temperature float32) {
```

Comment:
Do we need this method?

Reply:
It was added to show how we set up temperature or change temperature